tests/database_upgrade.rs

```diff
@@ -0,0 +1,213 @@
+#![allow(unused_imports)]
+#![allow(dead_code)]
+
+mod common;
+
+mod database_upgrade {
+	use super::common::docker::DockerContainer;
+	use super::common::expected::Expected;
+	use super::common::rest_client::RestClient;
+	use serde_json::Value as JsonValue;
+	use serial_test::serial;
+	use std::time::Duration;
+	use surrealdb::engine::any::{connect, Any};
+	use surrealdb::{Connection, Surreal};
+	use test_log::test;
+	use tracing::info;
+	use ulid::Ulid;
+
+	const CNX_TIMEOUT: Duration = Duration::from_secs(180);
+	const NS: &str = "test";
+	const DB: &str = "test";
+	const USER: &str = "root";
+	const PASS: &str = "root";
+
+	#[test(tokio::test(flavor = "multi_thread"))]
+	#[cfg(feature = "storage-rocksdb")]
+	#[serial]
+	async fn upgrade_test_1_0_1() {
+		// Start the docker instance
+		let (path, mut docker, client) = start_docker("1.0.1").await;
+
+		// Create the data set
+		create_data_on_docker(&client, "FTS", &DATA_FTS).await;
+
+		// Check the data set
+		check_data_on_docker(&client, "FTS", &CHECK_FTS).await;
+		check_data_on_docker(&client, "DB", &CHECK_DB).await;
+
+		// Stop the docker instance
+		docker.stop();
+
+		// Extract the database directory
+		docker.extract_data_dir(&path);
+
+		// Connect to a local instance
+		let db = new_local_instance(&path).await;
+
+		// Check that the data has properly migrated
+		check_migrated_data(&db, "DB", &CHECK_DB).await;
+		check_migrated_data(&db, "FTS", &CHECK_FTS).await;
+	}
+
+	#[test(tokio::test(flavor = "multi_thread"))]
+	#[cfg(feature = "storage-rocksdb")]
+	#[serial]
+	async fn upgrade_test_1_1_1() {
```

Comment:
Not a blocker for this PR, but probably we want to have tests for upgrades from respective 1.x.0 versions?
I thought the original issue was also about upgrading from v1.1.0, which is not covered specifically by any of the tests defined in this file currently.

Reply:
It makes sense. I am adding migrating from every patched version.
So we have:
1.0.0 -> 1.3
1.0.1 -> 1.3
1.1.0 -> 1.3
1.1.1 -> 1.3
1.2.0 -> 1.3
1.2.1 -> 1.3
Migration from minor versions (1.0.0 -> 1.0.1    1.1.0 -> 1.1.1) will be done in a different branch and PR